internal/controller/progressive/progressive.go

```diff
@@ -142,68 +142,56 @@ func ProcessResource(
 	}
 
 	// is there currently an "upgrading" child?
-	currentUpgradingChildDef, err := ctlrcommon.FindMostCurrentChildOfUpgradeState(ctx, rolloutObject, common.LabelValueUpgradeInProgress, nil, false, c)
+	currentUpgradingChildDef, err := ctlrcommon.FindMostCurrentChildOfUpgradeState(ctx, rolloutObject, common.LabelValueUpgradeInProgress, nil, true, c)
 	if err != nil {
 		return false, 0, err
 	}
 
 	// if there's a difference between the desired spec and the current "promoted" child, and there isn't already an "upgrading" definition, then create one and return
-	if promotedDifference && currentUpgradingChildDef == nil {
-		// Create it, first making sure one doesn't already exist by checking the live K8S API
-		currentUpgradingChildDef, err = ctlrcommon.FindMostCurrentChildOfUpgradeState(ctx, rolloutObject, common.LabelValueUpgradeInProgress, nil, true, c)
-		if err != nil {
-			return false, 0, fmt.Errorf("error getting %s: %v", currentUpgradingChildDef.GetKind(), err)
-		}
+	if promotedDifference {
 		if currentUpgradingChildDef == nil {
 			_, needRequeue, err := startUpgradeProcess(ctx, rolloutObject, existingPromotedChild, controller, c)
 			if needRequeue {
 				return false, common.DefaultRequeueDelay, err
 			} else {
 				return false, 0, err
 			}
-		}
-	}
 
-	// nothing to do (either there's nothing to upgrade, or we just created an "upgrading" child, and it's too early to start reconciling it)
-	if currentUpgradingChildDef == nil {
-		return true, 0, err
-	}
```

Comment:
this was the only place where we weren't previously calling the Live K8S API to get the upgrading child. 

This is not called very often actually, so we weren't saving much on Live K8S API calls anyway, and the code was more complicated for no good reason.